techlibs/ecp5/bram.txt

```diff
@@ -37,16 +37,78 @@ bram $__ECP5_DP16KD
   clkpol 2 3
 endbram
 
+# The syn_* attributes are described in:
+# https://www.latticesemi.com/-/media/LatticeSemi/Documents/Tutorials/AK/LatticeDiamondTutorial311.ashx
+attr_icase 1
+
 match $__ECP5_PDPW16KD
+	# implicitly requested RAM or ROM
+	attribute !syn_ramstyle syn_ramstyle=auto
+	attribute !syn_romstyle syn_romstyle=auto
+	attribute !ram_block
+	attribute !rom_block
+	attribute !logic_block
```

Comment:
nitpick: looks like tabs and spaces have been mixed here

Reply:
ah good catch, I fixed this for ice40 but forgot ecp5